tools/analyzer_plugin/lib/src/diagnostic/single_child_key.dart

```diff
@@ -21,7 +21,9 @@ class SingleChildWithKey extends ComponentUsageDiagnosticContributor {
     final parentMethodName = usage.node.thisOrAncestorOfType<MethodDeclaration>()?.name?.name;
 
     final parent = usage.node.parent;
-    if (parent is ListLiteral && (parent?.parent is! ReturnStatement)) {
+    // Being a list literal, the grandparent should be another component usage. Otherwise, the list may still
+    // easily be manipulated.
```

Comment:
Could we reword this comment? I'm not quite sure what this is saying, especially this part:

> Otherwise, the list may still easily be manipulated.

Reply:
Also, did you see this thread? https://workiva.slack.com/archives/CFM41D7C1/p1593102720005300?thread_ts=1593102203.002800&cid=CFM41D7C1

> luis.gomez  5 hours ago
> Iterable with 1 member missing key **does** throw a React runtime warning
>
>corwin.sheahan  5 hours ago
> IMO if the react runtime throws a warning for that, we probably should too :shrug:__

Sounds like we should still warn when the list only has a single child, which means we need to update this logic

Reply:
Yep! I can update the comment. Doesn't this achieve that goal though? I thought that's what this lint originally did and we needed to make it more focused

Reply:
Oh, wait, nevermind haha

Reply:
Updated @greglittlefield-wf! I renamed a bunch of stuff because I think this lint was now doing an entirely different thing than it originally was haha